File: PresentationTest/PopAdViewController.m
//
//  PopAdViewController.m
//  PresentationTest
//
//  Created by Minxin Guo on 5/9/16.
//  Copyright © 2016 aisle411. All rights reserved.
//

#import "PopAdViewController.h"

@interface PopAdViewController ()
@property (nonatomic, strong) UIImageView * imageView;
@property (nonatomic, strong) UIButton *closeButton;
@property (nonatomic, strong) UIView *containerView;
@end

@implementation PopAdViewController

- (instancetype)init
{
    self = [super initWithNibName:nil bundle:nil];
    if (self) {
        self.modalPresentationStyle = UIModalPresentationCustom;
        _imageView = [[UIImageView alloc] init];
        _closeButton = [[UIButton alloc] init];
        _containerView = [[UIView alloc] init];
        [self createUI];
    }
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    return self;
}

- (void)createUI {
    _containerView.translatesAutoresizingMaskIntoConstraints = NO;
    _containerView.backgroundColor = [[UIColor lightGrayColor] colorWithAlphaComponent:0.6];
    _containerView.layer.cornerRadius = 5.0;
    [self.view addSubview:_containerView];
    
    CGFloat bodyFontSize = [self.view bounds].size.width > 667.0 ? 20.0 : 16.0;
    _closeButton.translatesAutoresizingMaskIntoConstraints = NO;
    [_closeButton setTitle:@"Close" forState:UIControlStateNormal];
    _closeButton.layer.cornerRadius = 5.0;
    _closeButton.titleLabel.font = [UIFont systemFontOfSize:bodyFontSize];
    _closeButton.backgroundColor = [UIColor colorWithRed:116.0/255 green:190.0/255 blue:42.0/255 alpha:1.0];
    [_closeButton addTarget:self action:@selector(close:) forControlEvents:UIControlEventTouchUpInside];
    [_containerView addSubview:_closeButton];
    
    _imageView.translatesAutoresizingMaskIntoConstraints = NO;
    _imageView.contentMode = UIViewContentModeScaleAspectFit;
    _imageView.clipsToBounds = YES;
    [_containerView addSubview:_imageView];
    NSDictionary *views = @{@"containerView" : _containerView, @"closeButton" : _closeButton, @"imageView" : _imageView};
    [self.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[containerView]|" options:0 metrics:nil views:views]];
    [self.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[containerView]|" options:0 metrics:nil views:views]];
    [_containerView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-[imageView(300@250)]-|" options:0 metrics:nil views:views]];
    [_containerView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-50-[closeButton(100@750)]-50-|" options:NSLayoutFormatAlignAllCenterY metrics:nil views:views]];
    [_containerView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-30-[imageView]-20-[closeButton(40)]-|" options:NSLayoutFormatAlignAllCenterX metrics:nil views:views]];
}

- (void)setImage: (UIImage *)image {
    [_imageView setImage:image];
}

- (void)close:(id)sender {
    [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}
@end

File: PresentationTest/PopAdViewController.swift
//
//  PopAdViewController.swift
//  PresentationTest
//
//  Created by Minxin Guo on 5/5/16.
//  Copyright © 2016 aisle411. All rights reserved.
//

import UIKit

class PopAdViewController: UIViewController {
    
    private var closeButton : UIButton = UIButton()
    private var imageView : UIImageView = UIImageView()
    private var containerView: UIView = UIView()
    
    init() {
        super.init(nibName: nil, bundle: nil)
        modalPresentationStyle = UIModalPresentationStyle.Custom
        createUI()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    private func createUI() {
        containerView.translatesAutoresizingMaskIntoConstraints = false
        containerView.backgroundColor = UIColor.lightGrayColor().colorWithAlphaComponent(0.6)
        containerView.layer.cornerRadius = 5.0
        self.view.addSubview(containerView)
        
        let bodyFontSize: CGFloat = view.bounds.size.width > 667.0 ? 20.0 : 16.0
        closeButton.translatesAutoresizingMaskIntoConstraints = false
        closeButton.setTitle("Close", forState: .Normal)
        closeButton.layer.cornerRadius = 5.0
        closeButton.titleLabel!.font = UIFont.systemFontOfSize(bodyFontSize)
        closeButton.backgroundColor = UIColor(red: 116.0/255, green:190.0/255, blue:42.0/255, alpha:1.0)
        closeButton.addTarget(self, action: #selector(close(_:)), forControlEvents: UIControlEvents.TouchUpInside)
        containerView.addSubview(closeButton)
        
        imageView.translatesAutoresizingMaskIntoConstraints = false
        containerView.addSubview(imageView)
        
        let views: [String : AnyObject] = ["containerView" : containerView, "closeButton" : closeButton, "imageView" : imageView]
        view.addConstraints(NSLayoutConstraint.constraintsWithVisualFormat("H:|[containerView]|", options: [], metrics: nil, views: views))
        view.addConstraints(NSLayoutConstraint.constraintsWithVisualFormat("V:|[containerView]|", options: [], metrics: nil, views: views))
        containerView.addConstraints(NSLayoutConstraint.constraintsWithVisualFormat("H:|-[imageView(300@250)]-|", options: [], metrics: nil, views: views))
        containerView.addConstraints(NSLayoutConstraint.constraintsWithVisualFormat("H:|-50-[closeButton(100@750)]-50-|", options: [.AlignAllCenterY], metrics: nil, views: views))
        containerView.addConstraints(NSLayoutConstraint.constraintsWithVisualFormat("V:|-30-[imageView]-20-[closeButton(40)]-|", options: [.AlignAllCenterX], metrics: nil, views: views))
    }
    
    func setImage(image: UIImage) {
        self.imageView.image = image
    }
    
    func close(sender: UIButton) {
        presentingViewController!.dismissViewControllerAnimated(true, completion: nil)
    }
    
}
